### PR TITLE
chapter 3 Mostly Static Pages

### PR DESCRIPTION
**Pull request by daniadore**
Ready

### CODE REVIEW

**Comment by daniadore** [comment]
Ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/* universal */
+
+body {
+ padding-top: 60px;
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss:4]: indent 2 space please
have you setting sublime text ?

File: app/assets/stylesheets/static_pages.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/static_pages.scss:2]: remove this file because it not use

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+   def full_title page_title = ""
+ base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb:3]: wrong indent

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+ <div class="container">
+ <%= link_to "sample app", "#", id: "logo" %>
+ <nav>
+ <ul class="nav navbar-nav navbar-right">
```
Comment (ngocvt-0484) [inline app/views/layouts/_header.html.erb:5]: please format your html
it is indenting wrong

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -11,6 +11,10 @@
   </head>
 
   <body>
-    <%= yield %>
-  </body>
+<%= render "layouts/header" %>
+ <div class="container">
+ <%= yield %>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb:16]: same

File: app/views/static_pages/help.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<%= provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+ Get help on the Ruby on Rails Tutorial at the
+ <a href="http://www.railstutorial.org/help">Rails Tutorial
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb:6]: please use link_to

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
```
Comment (ngocvt-0484) [inline app/views/static_pages/home.html.erb:3]: please apply i18n for all text on html.erb

File: test/controllers/static_pages_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
+
+class StaticPagesControllerTest < ActionDispatch::IntegrationTest
```
Comment (ngocvt-0484) [inline test/controllers/static_pages_controller_test.rb:3]: test don't need to do
delete it please

Comment (daniadore) [reply test/controllers/static_pages_controller_test.rb:3]: sorry but i dont know which one to delete ?
image

**Comment by daniadore** [comment]
Ready

**Comment by daniadore** [comment]
Ready

File: config/locales/en.yml [hunk 9]
```diff
@@ -1,33 +1,11 @@
...
-#
-# Instead, surround them with single quotes.
-#
-# en:
-#   'true': 'foo'
-#
-# To learn more, please read the Rails Internationalization guide
-# available at https://guides.rubyonrails.org/i18n.html.
-
+---
```
Comment (thaodtd-0665) [inline config/locales/en.yml:1]: remove this line
```suggestion
```

File: test/controllers/static_pages_controller_test.rb (hunk 8 above)
Comment (thaodtd-0665) [reply test/controllers/static_pages_controller_test.rb:3]: you should delete folder ../test/controllers

File: app/views/static_pages/home.html.erb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<%= provide :title, "Home" %>
```
Comment (thaodtd-0665) [inline app/views/static_pages/home.html.erb:1]: using i18n for text in this file html
```suggestion
<%= provide :title, t(:home) %>
```

File: app/controllers/static_pages_controller.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
```
Comment (huongnt-2545) [inline app/controllers/static_pages_controller.rb:2]: if method is empty

```suggestion
  # good
  def home; end
  
  def help; end
```

File: app/helpers/application_helper.rb [hunk 12]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+   base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb:3]: indent = 2 spaces

```suggestion
    base_title = "Ruby on Rails Tutorial Sample App"
```

File: app/views/layouts/_header.html.erb [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "sample app", "#", id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home", "#" %></li>
```
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb:6]: apply I18n to translate text like `"Home", "Help", etc.`

File: config/locales/ja.yml [hunk 14]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (huongnt-2545) [inline config/locales/ja.yml:1]: remove this line

**Comment by daniadore** [comment]
Ready

File: app/helpers/application_helper.rb [hunk 15]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb:3]: apply I18n for this text `"Ruby on Rails Tutorial Sample App"`

File: app/views/visitors/index.html.erb [hunk 16]
```diff
@@ -0,0 +1,3 @@
+<h1><% t :greetings %></h1>
+<h4>Default: <%= l Date.today %></h4>
```
Comment (huongnt-2545) [inline app/views/visitors/index.html.erb:2]: I18n for text `Default:`, `Short:`

**Review by huongnt-2545 — APPROVED** [review]
merged
I have some small comments relate to I18n, you can fix it on the next pull request
You can merge this pull request

**Pull request merged**